engine/src/main/java/org/terasology/rendering/dag/ShadowMapNode.java

```diff
@@ -0,0 +1,146 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.GL11;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.context.Context;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.backdrop.BackdropProvider;
+import org.terasology.rendering.cameras.Camera;
+import org.terasology.rendering.cameras.OrthographicCamera;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.primitives.ChunkMesh;
+import org.terasology.rendering.world.RenderQueuesHelper;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.rendering.world.WorldRendererImpl;
+import static org.lwjgl.opengl.GL11.*;
+import static org.terasology.rendering.opengl.OpenGLUtils.*;
+
+
+/**
+ * Diagram of this node can be viewed from:
+ * TODO: move diagram to the wiki when this part of the code is stable
+ * - https://docs.google.com/drawings/d/13I0GM9jDFlZv1vNrUPlQuBbaF86RPRNpVfn5q8Wj2lc/edit?usp=sharing
+ */
+public class ShadowMapNode implements Node {
+    private static final int SHADOW_FRUSTUM_BOUNDS = 500;
+
+
+    public Camera camera = new OrthographicCamera(-SHADOW_FRUSTUM_BOUNDS, SHADOW_FRUSTUM_BOUNDS, SHADOW_FRUSTUM_BOUNDS, -SHADOW_FRUSTUM_BOUNDS);
+    private RenderQueuesHelper renderQueues;
+    private Context context;
+    private Material shadowMapShader;
+    private FBO shadowMap;
+
+    // TODO: every node proposes its modifiable configuration?
+    private RenderingConfig renderingConfig;
+
+    private Camera playerCamera;
+    // FIXME: remove reference to WorldRendererImpl
+    private WorldRenderer worldRenderer;
+    private BackdropProvider backdropProvider;
+
+
+    // FIXME: unnecessary arguments must be eliminated
+    public ShadowMapNode(Context context, Material shadowMapShader, Camera playerCamera) {
+        this.context = context;
+        this.shadowMapShader = shadowMapShader;
+        this.playerCamera = playerCamera;
+        this.worldRenderer = context.get(WorldRenderer.class);
+        this.backdropProvider = context.get(BackdropProvider.class);
+        this.renderingConfig = context.get(Config.class).getRendering();
+        this.shadowMap = context.get(FrameBuffersManager.class).getFBO("sceneShadowMap");
+        this.renderQueues = context.get(RenderQueuesHelper.class);
+    }
+
+    @Override
+    public void update(float deltaInSeconds) {
+        this.shadowMap = context.get(FrameBuffersManager.class).getFBO("sceneShadowMap");
```

Comment:
Of course this doesn't mean you should just cut and paste everything into process(). I.e. all camera positioning stuff should go in its private method.